packages/block-directory/src/store/controls.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * External dependencies
+ */
+import { forEach } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { createRegistryControl } from '@wordpress/data';
+import wpApiFetch from '@wordpress/api-fetch';
+
+/**
+ * Calls a selector using the current state.
+ *
+ * @param {string} storeName    Store name.
+ * @param {string} selectorName Selector name.
+ * @param {Array}  args         Selector arguments.
+ *
+ * @return {Object} Control descriptor.
+ */
+export function select( storeName, selectorName, ...args ) {
+	return {
+		type: 'SELECT',
+		storeName,
+		selectorName,
+		args,
+	};
+}
+
+/**
+ * Calls a dispatcher using the current state.
+ *
+ * @param {string} storeName      Store name.
+ * @param {string} dispatcherName Dispatcher name.
+ * @param {Array}  args           Selector arguments.
+ *
+ * @return {Object} Control descriptor.
+ */
+export function dispatch( storeName, dispatcherName, ...args ) {
+	return {
+		type: 'DISPATCH',
+		storeName,
+		dispatcherName,
+		args,
+	};
+}
+
+/**
+ * Trigger an API Fetch request.
+ *
+ * @param {Object} request API Fetch Request Object.
+ *
+ * @return {Object} Control descriptor.
+ */
+export function apiFetch( request ) {
+	return {
+		type: 'API_FETCH',
+		request,
+	};
+}
+
+/**
+ * Loads JavaScript
+ *
+ * @param {Array}    asset   The url for the JavaScript.
+ * @param {Function} onLoad  Callback function on success.
+ * @param {Function} onError Callback funciton on failure.
+ */
+const loadScript = ( asset, onLoad, onError ) => {
+	if ( ! asset ) {
+		return;
+	}
+	const existing = document.querySelector( `script[src="${ asset.src }"]` );
```

Comment:
The biggest challenge here is that it doesn't ensure that all dependencies of the script are loaded. It probably won't be the case on the edit post page though unless someone uses one of the WordPress script handles which are not used on this page.

Reply:
Ah, great point. Let's think about this some more and address it in a follow-up PR.

Reply:
**Adding an update:**
This problem could potentially be addressed here:
- [Wordpress Core Ticket](https://core.trac.wordpress.org/ticket/48654 ): 
- Gutenberg Ticket: https://github.com/WordPress/gutenberg/issues/21738